src/attribute.rs

```diff
@@ -246,7 +246,13 @@ impl<'n, 'f> NtfsAttribute<'n, 'f> {
 
         let start = self.offset + self.name_offset() as usize;
         let end = start + self.name_length();
-        let string = NtfsString(&self.file.record_data()[start..end]);
+        let string = NtfsString(&self.file.record_data().get(start..end).ok_or(
+            NtfsError::InvalidAttributeNameRange {
+                position: self.position(),
+                range: start..end,
+                size: self.file.record_data().len(),
+            },
+        )?);
```

Comment:
Congratulations! You managed to break the case where I promised extensive validation and safe indexing inside `self.file.record_data()` :)

Turns out that while I checked that `name_offset` and `name_length` are within `attribute_length`, I never checked `attribute_length` itself.
It must be at least the size of the `NtfsAttributeHeader` structure and at most the remaining file record size.
Otherwise, every indexing into `self.file.record_data()` inside `NtfsAttribute` may potentially panic.

Currently thinking about the best location to validate `attribute_length`.
When that is done, we can revert commits `c3c871e` and `e01fb63`. We would then also return a way more precise error, because the root cause for these errors is not in the `name` or `resident_value` methods, but instead in the invalid `attribute_length`.

Reply:
Ok i got it, thanks for fixing it then 😄 , maybe a good example that validation functions are good for error precision but not that much for crash protection 🤔